src/future/mod.rs

```diff
@@ -957,3 +971,20 @@ pub trait FutureFrom<T>: Sized {
     /// Consume the given value, beginning the conversion.
     fn future_from(T) -> Self::Future;
 }
+
+/// A trait for types which can spawn fresh futures.
+///
+/// This trait is typically implemented for "executors", or those types which
+/// can execute futures to completion. Futures passed to `Spawn::spawn`
+/// typically get turned into a *task* and are then driven to completion.
+///
+/// On spawn, the executor takes ownership of the future and becomes responsible
+/// to call `Future::poll()` whenever a readiness notification is raised.
+pub trait Spawn<F: Future<Item = (), Error = ()>> {
+    /// Spawns a future to run on this `Spawn`.
+    ///
+    /// This function will return immediately, and schedule the future `future`
+    /// to run on `self`. The details of scheduling and execution are left to
+    /// the implementations of `Spawn`.
+    fn spawn(&self, future: F);
```

Comment:
This function should return `Result<(), Error<F>>` to allow the executor to return the future if it fails to spawn it.

The docs should say that returning the future on error is "best effort" and returning `Ok` is not a guarantee that the future will be executed.